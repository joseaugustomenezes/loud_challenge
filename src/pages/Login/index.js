import React from "react";
import { useDispatch, useSelector } from "react-redux";
import Container from "react-bootstrap/Container";
import { Formik } from "formik";
import Form from "react-bootstrap/Form";
import { Link } from "react-router-dom";

import LoadingButton from '../../components/LoadingButton';
import Toast from "../../components/Toast";
import { login } from "../../store/actions";

const Login = () => {
  const dispatch = useDispatch();
  const user = useSelector((state) => state.user);

  return (
    <div style={{ possition: "relative" }}>
      <Toast message={user?.error?.message} status={user?.error?.status} />
      <Container>
        <h1>Login</h1>
        <hr />
        <Formik
          initialValues={{ email: "", username: "", password: "" }}
          validate={(values) => {
            const errors = {};
            if (!values.username) errors.username = "Campo obrigat칩rio";
            if (!values.password) errors.password = "Campo obrigat칩rio";
            return errors;
          }}
          onSubmit={(values) => {
            dispatch(login(values));
          }}
        >
          {({
            values,
            handleSubmit,
            handleChange,
            handleBlur,
            touched,
            errors,
          }) => (
            <Form onSubmit={handleSubmit}>
              <Form.Group>
                <Form.Label>Usu치rio</Form.Label>
                <Form.Control
                  type="text"
                  name="username"
                  id="username"
                  placeholder="Insira o usu치rio"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.username}
                  isInvalid={touched.username && errors.username}
                />
                <Form.Control.Feedback type="invalid">
                  {errors.username}
                </Form.Control.Feedback>
              </Form.Group>
              <Form.Group>
                <Form.Label>Senha</Form.Label>
                <Form.Control
                  type="password"
                  name="password"
                  ogin
                  id="password"
                  placeholder="Insira a senha"
                  onChange={handleChange}
                  onBlur={handleBlur}
                  value={values.password}
                  isInvalid={touched.password && errors.password}
                />
                <Form.Control.Feedback type="invalid">
                  {errors.password}
                </Form.Control.Feedback>
              </Form.Group>
              <LoadingButton loading={user?.loading} type="submit">Entrar</LoadingButton>
              <span> ou </span>
              <Link to="/signup">Cadastar</Link>
            </Form>
          )}
        </Formik>
      </Container>
    </div>
  );
};

export default Login;
