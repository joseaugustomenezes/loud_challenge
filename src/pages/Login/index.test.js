import React from "react";
import { mount } from "enzyme";
import configureStore from "redux-mock-store";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";

import Login from "./index";

describe("<Login />", () => {
  it("should render without any errors", () => {
    const mockStore = configureStore();
    const component = mount(
      <Router>
        <Provider store={mockStore({})}>
          <Login />
        </Provider>
      </Router>
    );
    expect(component).toMatchSnapshot();
    component.unmount();
  });

  it("should render empty inputs", async () => {
    const mockStore = configureStore();
    const component = mount(
      <Router>
        <Provider store={mockStore({})}>
          <Login />
        </Provider>
      </Router>
    );
    await component.find('MyInnerForm').props().submitForm();
    expect(component.find('MyInnerForm').update().find('.invalid-feedback').at(0).contains('Campo obrigatório')).toBeTruthy();
    expect(component.find('MyInnerForm').update().find('.invalid-feedback').at(1).contains('Campo obrigatório')).toBeTruthy();
    component.unmount();
  });
});
