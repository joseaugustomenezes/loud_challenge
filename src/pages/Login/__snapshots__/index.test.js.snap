// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should render without any errors 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <Login>
        <div
          style={
            Object {
              "possition": "relative",
            }
          }
        >
          <Toast>
            <div
              style={
                Object {
                  "position": "absolute",
                  "right": 20,
                  "top": 10,
                }
              }
            >
              <Toast
                autohide={true}
                delay={3000}
                onClose={[Function]}
                show={false}
              >
                <Fade
                  appear={false}
                  in={false}
                  mountOnEnter={false}
                  timeout={300}
                  unmountOnExit={true}
                >
                  <Transition
                    addEndListener={[Function]}
                    appear={false}
                    enter={true}
                    exit={true}
                    in={false}
                    mountOnEnter={false}
                    onEnter={[Function]}
                    onEntered={[Function]}
                    onEntering={[Function]}
                    onExit={[Function]}
                    onExited={[Function]}
                    onExiting={[Function]}
                    timeout={300}
                    unmountOnExit={true}
                  />
                </Fade>
              </Toast>
            </div>
          </Toast>
          <Container
            fluid={false}
          >
            <div
              className="container"
            >
              <h1>
                Login
              </h1>
              <hr />
              <Formik
                initialValues={
                  Object {
                    "password": "",
                    "username": "",
                  }
                }
                onSubmit={[Function]}
                validate={[Function]}
              >
                <MyInnerForm
                  dirty={false}
                  errors={Object {}}
                  getFieldHelpers={[Function]}
                  getFieldMeta={[Function]}
                  getFieldProps={[Function]}
                  handleBlur={[Function]}
                  handleChange={[Function]}
                  handleReset={[Function]}
                  handleSubmit={[Function]}
                  initialErrors={Object {}}
                  initialTouched={Object {}}
                  initialValues={
                    Object {
                      "password": "",
                      "username": "",
                    }
                  }
                  isSubmitting={false}
                  isValid={true}
                  isValidating={false}
                  registerField={[Function]}
                  resetForm={[Function]}
                  setErrors={[Function]}
                  setFieldError={[Function]}
                  setFieldTouched={[Function]}
                  setFieldValue={[Function]}
                  setFormikState={[Function]}
                  setStatus={[Function]}
                  setSubmitting={[Function]}
                  setTouched={[Function]}
                  setValues={[Function]}
                  submitCount={0}
                  submitForm={[Function]}
                  touched={Object {}}
                  unregisterField={[Function]}
                  validateField={[Function]}
                  validateForm={[Function]}
                  validateOnBlur={true}
                  validateOnChange={true}
                  validateOnMount={false}
                  values={
                    Object {
                      "password": "",
                      "username": "",
                    }
                  }
                >
                  <Form
                    inline={false}
                    onSubmit={[Function]}
                  >
                    <form
                      className=""
                      onSubmit={[Function]}
                    >
                      <FormGroup>
                        <div
                          className="form-group"
                        >
                          <FormLabel
                            column={false}
                            srOnly={false}
                          >
                            <label
                              className="form-label"
                            >
                              Usuário
                            </label>
                          </FormLabel>
                          <FormControl
                            id="username"
                            name="username"
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Insira o usuário"
                            type="text"
                            value=""
                          >
                            <input
                              className="form-control"
                              id="username"
                              name="username"
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Insira o usuário"
                              type="text"
                              value=""
                            />
                          </FormControl>
                          <Feedback
                            type="invalid"
                          >
                            <div
                              className="invalid-feedback"
                            />
                          </Feedback>
                        </div>
                      </FormGroup>
                      <FormGroup>
                        <div
                          className="form-group"
                        >
                          <FormLabel
                            column={false}
                            srOnly={false}
                          >
                            <label
                              className="form-label"
                            >
                              Senha
                            </label>
                          </FormLabel>
                          <FormControl
                            id="password"
                            name="password"
                            ogin={true}
                            onBlur={[Function]}
                            onChange={[Function]}
                            placeholder="Insira a senha"
                            type="password"
                            value=""
                          >
                            <input
                              className="form-control"
                              id="password"
                              name="password"
                              ogin={true}
                              onBlur={[Function]}
                              onChange={[Function]}
                              placeholder="Insira a senha"
                              type="password"
                              value=""
                            />
                          </FormControl>
                          <Feedback
                            type="invalid"
                          >
                            <div
                              className="invalid-feedback"
                            />
                          </Feedback>
                        </div>
                      </FormGroup>
                      <LoadingButton
                        type="submit"
                      >
                        <Button
                          active={false}
                          disabled={false}
                          type="submit"
                          variant="primary"
                        >
                          <button
                            className="btn btn-primary"
                            disabled={false}
                            type="submit"
                          >
                            Entrar
                          </button>
                        </Button>
                      </LoadingButton>
                      <span>
                         ou 
                      </span>
                      <Link
                        to="/signup"
                      >
                        <LinkAnchor
                          href="/signup"
                          navigate={[Function]}
                        >
                          <a
                            href="/signup"
                            onClick={[Function]}
                          >
                            Cadastar
                          </a>
                        </LinkAnchor>
                      </Link>
                    </form>
                  </Form>
                </MyInnerForm>
              </Formik>
            </div>
          </Container>
        </div>
      </Login>
    </Provider>
  </Router>
</BrowserRouter>
`;
